### PR TITLE
Allow resty backend timeout to be configurable

### LINKED ISSUE #3635: [Proposal] Allowing Resty HTTP Timeout to be configurable
https://github.com/sensu/sensu-go/blob/master/cli/client/client.go#L43

The question/proposal is centered around the above code ^^.  Would a PR be acceptable to
* allow that timeout to be configurable at all?
* allow that timeout to be configurable from the cli?
* allow that timeout configurable item to be persisted in the configuration file

## Possible Implementation

Draft PR here https://github.com/naemono/sensu-go/pull/1

## Context

In heavily used sensu-go installations, sometimes 15 seconds isn't quite enough time to get a response.

### PR DESCRIPTION
**Pull request by naemono**
## What is this change?

Allow resty backend timeout to be configurable

## Why is this change necessary?

Closes #3635 

The timeout isn't configurable, as gets triggered on highly used installations.

## Does your change need a Changelog entry?

Yes, the changelog was updated in the "Unreleased" section

## Have you reviewed and updated the documentation for this change? Is new documentation required?

Documentation likely needs to be updated about this change.  Could someone point me to the right location?

## How did you verify this change?

Build sensuctl binary locally, and verified the setting was adjusted, and persisted in configuration file

## Is this change a patch?

No

### CODE REVIEW

File: cli/client/config/mock.go [hunk 1]
```diff
@@ -38,6 +40,14 @@ func (m *MockConfig) Namespace() string {
 	return args.String(0)
 }
 
+// Timeout mocks the timeout config
+func (m *MockConfig) Timeout() time.Duration {
+	args := m.Called()
+	timeoutString := args.String(0)
+	d, _ := time.ParseDuration(timeoutString)
+	return d
```
Comment (echlebek) [inline cli/client/config/mock.go]: What you can do here is 
```
return args.Get(0).(time.Duration)
```

File: cli/client/testing/mock_config.go [hunk 2]
```diff
@@ -43,6 +45,14 @@ func (m *MockConfig) TrustedCAFile() string {
 	return args.String(0)
 }
 
+// Timeout mocks the timeout config
+func (m *MockConfig) Timeout() time.Duration {
+	args := m.Called()
+	timeoutString := args.String(0)
+	d, _ := time.ParseDuration(timeoutString)
+	return d
```
Comment (echlebek) [inline cli/client/testing/mock_config.go]: Similarly here, you can use 

```
args.Get(0).(time.Duration)
```

File: cli/client/config/mock.go (hunk 1 above)
Comment (naemono) [reply cli/client/config/mock.go]: 👍

File: cli/client/testing/mock_config.go (hunk 2 above)
Comment (naemono) [reply cli/client/testing/mock_config.go]: 👍

**Comment by naemono** [comment]
@echlebek these issues should be resolved now with latest commit.

Please be aware of https://github.com/etcd-io/bbolt/issues/187 as I was running go 1.14, and ran directly into this when running unit tests.  Mass failure.

**Comment by echlebek** [comment]
To be merged following the 5.19.0 release.

**Pull request merged**